### ISSUE TITLE
disable KeyboardShrinksView on iOS.

### ISSUE THREAD

**Issue by fidoboy** [AppGyver/steroids#840]
When keyboard is being displayed the webView is being stretched and it looks very ugly. There is no way to disable this behavior? (tested on iPad Air with iOS 8.1.3). This also happens if keyboard is being displayed from drawer where the center view is being stretched.

Please :pray: solve this issue!! :)

**Comment by fidoboy** [comment]
NOTE: This doesn't happen on iPhone with iOS 7.1.2. The webView is not being stretched

**Comment by adamgoose** [comment]
This used to be supported by a config option, but no longer does. Please bring it back!

**Comment by bFlood** [comment]
+1

**Comment by harsaharsa** [comment]
The old Cordova keyboard settings stopped working on iOS 8, and we transitioned to https://github.com/driftyco/ionic-plugins-keyboard/ – which unfortunately doesn't provide the "KeyboardShrinksView" option.

Your feature request is heard, though, we'll put it on the backlog!

**Comment by fidoboy** [comment]
Ok, I'll try to fix the plugin then... i'll report back my progress here

**Comment by fidoboy** [comment]
Please, take a look at these pull request: https://github.com/apache/cordova-plugins/pulls it seems that they solves the problem on iOS. The first one does seems to work on iOS 8 and the second one fixes the problem with accessory bar. I can't add them into Steroids plugin list because the ionic keyboard plugin is already bundled. @harsaharsa can you please test it and replace it if they works?

**Comment by fidoboy** [comment]
The situation is very simple: the current keyboard plugin from ionic, doesn't work with iOS 8.x (and iPad) so it could be better to remove it from the build and let users to add what they want, this way you could experiment and search the one that fits best into your needs. With current plugin from ionic, if there is any input field at bottom, it's overlapped by the keyboard and you can see or reach it. It works however on iPhone and using the simulator (also using the iPad simulator curiously, but not with the real one)

**Comment by harsaharsa** [comment]
Thanks for the research @fidoboy, I'll bring this up with the iOS team!

**Comment by harsaharsa** [comment]
In iOS `v4.0.4-edge2`, available now from the Build Service, the Ionic keyboard plugin is removed. While testing on an iPad, this produced as-expected default behavior:

When focusing an input field, the keyboard comes up and the WebView is scrolled so that the input field is visible. However, the WebView height itself is not resized, meaning that if you have an input field at the top of the page and a `bottom: 0` absolute positioned element, the latter will be "under" the keyboard. Since on iOS8 the keyboard is translucent, the rest of the WebView shows through. A non-native option would be to resize the HTML document via JS to the correct size when the input is focused.

This now opens up other keyboard plugins to be included to fix this issue.

**Comment by adamgoose** [comment]
Unfortunately, this still changes the viewport height. :(

**Comment by fidoboy** [comment]
Cool @harsaharsa ! Now we are talking... :D I'll try to fix the plugin to work with iOS. I'm not sure however about Android, because the issue seems to be more serious...

**Comment by harsaharsa** [comment]
@adamgoose yup – I didn't find any actively maintained plugins that would let you define how the viewport changes height when the keyboard comes up. :( If @fidoboy comes up with something, we're happy to promote it!

**Comment by harsaharsa** [comment]
While this issue is still present, please see the new native input plugin at https://github.com/appgyver/input-plugin for a solution where you need the input field attached to the keyboard.
